pkg/metrics/metrics.go

```diff
@@ -78,41 +90,83 @@ func NewBuilder() MetricsBuilder {
 type metricsBuilder struct{}
 
 func (mb *metricsBuilder) NewClient(c client.Client) (Metrics, error) {
-	promHost, err := getPromHost(c)
+	promEndpoint, err := Endpoint(c, MonitoringNS, promApp, "web")
 	if err != nil {
 		return nil, err
 	}
 
-	token, err := getPrometheusToken(c)
+	token, err := prometheusToken(c)
 	if err != nil {
 		return nil, err
 	}
 
+	useRoutes := config.UseRoutes()
+	tlsConfig := &tls.Config{}
+
+	if !useRoutes {
+		tlsConfig, err = TLSConfig(c)
+		if err != nil {
+			return nil, err
+		}
+	}
+
 	return &Counter{
-		promHost: *promHost,
+		promEndpoint: promEndpoint,
 		promClient: http.Client{
 			Transport: &prometheusRoundTripper{
 				token: *token,
+				tls:   tlsConfig,
 			},
 		},
 	}, nil
 }
 
 type prometheusRoundTripper struct {
 	token string
+	tls   *tls.Config
+}
+
+func TLSConfig(c client.Client) (*tls.Config, error) {
```

Comment:
Should better explain the function if it is public

Reply:
Done, also updated to MonitoringTLSConfig